pages/docs/reference/fun-interfaces.md

```diff
@@ -0,0 +1,101 @@
+---
+type: doc
+layout: reference
+title: "Functional interfaces (SAM interfaces)"
+---
+
+# Functional (SAM) interfaces
```

Comment:
I propose adding a note somewhere to state that this functionality is available only since Kotlin 1.4

Reply:
Also, let's put (SAM) at the end: Functional Interfaces (SAM)

Reply:
> I propose adding a note somewhere to state that this functionality is available only since Kotlin 1.4

These docs will be published after 1.4 release. Our docs cover the latest version of Kotlin so we don't need to add _since Kotlin 1.4_. 
Also I would avoid using versions in documentation. It's ok to use them in What's new (I haven't prepared it yet). Our documentation already includes _since 1.1, since 1.2, since 1.3_. It looks very strange and is a bad practice (what's the purpose?). We will remove these _since_ when migrating to WH.

Docs describe the latest Kotlin version so our goal is to motivate people use it earlier. What's new covers when a specific feature was added consistently for all features.

Reply:
> Also, let's put (SAM) at the end: Functional Interfaces (SAM)

I had this initially but it was not gramatically correct. SAM is not the same as Functional interfaces, right?. _Functional interfaces (SAM interfaces)_ looks a bit long. I will consult with our proofreaders when I send them this content to proofread.